controllers/userController.js

```diff
@@ -1,28 +1,248 @@
+      })
+      let followings = user.Followings
+      followings = followings.map(following => ({
+        followingId: following.id,
+        ...Object.fromEntries(Object.entries(following.dataValues).slice(1, 5)),
+        followshipCreatedAt: following.Followship.createdAt,
+        isFollowed: helpers.getUser(req).Followings.map(following => following.id).includes(following.id)
+      }))
+      followings = followings.sort((a, b) => b.followshipCreatedAt - a.followshipCreatedAt)
+      return res.json(followings)
+    }).catch(next)
   },
-  readFollowings: (req, res) => {
 
+  readFollowers: (req, res, next) => {
+    const id = Number(req.params.id)
+    User.findByPk(id, {
+      include: [{
+        model: User,
+        as: 'Followers',
+        attributes: ['id', 'account', 'name', 'avatar', 'introduction']
+      }],
+    }).then(user => {
+      if (!user) return res.status(404).json({
+        message: `this user(id: ${id}) do not exist!`
+      })
+      let followers = user.Followers
+      followers = followers.map(follower => ({
+        followerId: follower.id,
+        ...Object.fromEntries(Object.entries(follower.dataValues).slice(1, 5)),
+        followshipCreatedAt: follower.Followship.createdAt,
+        isFollowed: helpers.getUser(req).Followings.map(following => following.id).includes(follower.id)
+      }))
+      followers = followers.sort((a, b) => b.followshipCreatedAt - a.followshipCreatedAt)
+      return res.json(followers)
+    }).catch(next)
   },
-  readFollowers: (req, res) => {
 
+  readTopUsers: (req, res, next) => {
+    User.findAll({
+      where: { role: 'user' },
+      attributes: ['id', 'account', 'name', 'avatar'],
+      include: [{
+        model: User,
+        as: 'Followers',
+        attributes: ['id']
+      }],
+    }).then(users => {
+      users = users.map(user => ({
+        ...Object.fromEntries(Object.entries(user.dataValues).slice(0, 4)),
+        followersCount: user.Followers.length,
+        isFollowed: helpers.getUser(req).Followings.map(following => following.id).includes(user.id)
+      }))
+      users = users.sort((a, b) => b.followersCount - a.followersCount)
+      users = users.slice(0, 10)
+      return res.json(users)
+    }).catch(next)
   },
-  updateUser: (req, res) => {
 
+  updateUser: (req, res, next) => {
+    const id = Number(req.params.id)
+    const update = req.body
+    const files = req.files || {}
+
+    // if update account
+    if (Object.keys(update).includes('account')) {
+      if (!update.account || !update.name || !update.email) {
+        return res.status(400).json({
+          status: 'failure',
+          message: 'account, name, email are required'
+        })
+      }
+      if (update.password !== update.checkPassword) {
+        return res.status(409).json({
+          status: 'failure',
+          message: 'Password & checkPassword are different!'
+        })
+      }
+      return findAndUpdate(res, next, id, update)
+    }
```

Comment:
```suggestion
 if (update.password !== update.checkPassword) {
        return res.status(409).json({
          status: 'failure',
          message: 'Password & checkPassword are different!'
        })
      }
      return findAndUpdate(res, next, id, update)
    }
```
於更改密碼的部分，目前設定新密碼以及確認新密碼，
是否要加入一個確認舊密碼的欄位來作二次驗證的動作。

Reply:
有幾點跟你確認：
1⃣️  單元測試就是我發的 PR message，想確認你認為「單元測試」的定義？
2⃣️  Coding style 一致的部分是指 `userObj` 嗎？按前面說明，僅一個 `Object` 不需要 `.map()` 展開
3⃣️  package看似安裝了很多套件，不確定每個的功能：這點在這個 commit [feat: add PUT /api/users/:id to update user's all info](https://github.com/klkuocx/twitter-api-2020/pull/17/commits/2bdb40c25323b1dde8e4407524798278c3d7468e) 有提到細節：
> - add multer to get req files
- add imgur to upload images on cloud and get their link
- add assert to check res of test

4⃣️  這兩點需要多解釋一點，看不太懂⋯⋯
> 潛水user與不存在使用者跑出同樣的錯誤訊息。
> 沒有排除account獨一無二的設定。

---

> 於更改密碼的部分，目前設定新密碼以及確認新密碼，是否要加入一個確認舊密碼的欄位來作二次驗證的動作。

這點超讚！確實是安全性比較高的體驗，那這要跟前端提一下，目前沒有這個設定！